### ISSUE TITLE
Dynamic methods invocation via '::' - Problem with ClassName::dynamicMethod()

### ISSUE THREAD

**Issue by iakoder** [kalessil/phpinspectionsea#1008]
First, thank you for reporting a bug or making a request. That takes time and we appreciate that!

Let's start with common details:
 
| Subject    | Details                                                  |
| :--------- | :------------------------------------------------------- |
| Issue type | Bug/False-Positive     |
| IDE        | PhpStorm, 2018.2.3 EAP   182.4323.23                       |
| Plugin     | Php Inspections (EA Extended), 3.0.5.1 |

The Dynamic methods invocation via '::' inspection doesn't catch dynamic methods invoked via ClassName::dynamicMethod() but it does with methods invoked via the self:: form.

```php
<?php

/**
 * Class MyClass
 */
class MyClass {

    /**
     *
     */
    public function dynamicMethod() {

    }

    /**
     *
     */
    public function anotherMethod() {

        
        // Why isn't this caught by Dynamic methods invocation via '::'
        MyClass::dynamicMethod();

    }

    public function aThirdMethod() {

        self::dynamicMethod();

    }

}
```

PhpStorm 2018.2.3 EAP
Build PS-182.4323.23, built on August 29, 2018
Licensed to Intacct / Dan Wilks
Subscription is active until June 23, 2019
JRE: 1.8.0_152-release-1248-b8 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.13.6
Notes

- Bugs fixed with the highest priority, so features might be postponed. But it shouldn't stop you, right?
- The project support is powered by community funding via [Patreon](https://www.patreon.com/kalessil)

**Comment by iakoder** [comment]
See also https://youtrack.jetbrains.com/issue/WI-43380

**Comment by kalessil** [comment]
Not sure why, will check. Thank you for the hands on!

**Comment by kalessil** [comment]
@iakoder : checked, that a new pattern. I'll leave the ticket in the current milestone, but it has to wait when open bugs are resolved.

**Comment by kalessil** [comment]
Implemented!
